### DISCUSSION TITLE
Five-year check-in for the Startup Founder’s Guide to Analytics

### DISCUSSION THREAD

**Discussion by dbeatty10** [dbt-labs/docs.getdbt.com#1366] [Archive]
Tristan wrote the [The Startup Founder’s Guide to Analytics](https://thinkgrowth.org/the-startup-founders-guide-to-analytics-1d2176f20ac1) five years ago, and the modern data stack (MDS) has continued to take shape since then.

I'm wondering if any of the recommendations for each stage has changed since then?

**Comment by krevitt** [comment]
Such a great topic.  My hunch is that earlier-stage startups have become much more sophisticated about how they do this stuff, because a single person doing data work just has a lot higher leverage / better tooling than they did when this was written.  

Thinking primarily about stuff like data activation / reverse ETL, and how data becoming operationalized *also* forces you to do stuff like adopt data testing / version control earlier than you would when all of your earlier data analysis was done in spreadsheets.

**Reply by adrienneyager** [reply]
> My hunch is that earlier-stage startups have become much more sophisticated about how they do this stuff, because a single person doing data work just has a lot higher leverage / better tooling than they did when this was written.

I strongly agree with this sentence, @krevitt! When I was starting up an analytics discipline at Torch in 2019, the bar to entry for setting up a warehouse, event tracking, testing, etc. was low enough that a single data person could set up large portions of the MDS quickly and easily.

**Comment by adrienneyager** [comment]
As someone who used this as a resource not too long ago to go through all of the listed stages, there are some sections I'd love to tweak:
* Implement data testing and documentation as soon as possible. This is important for maintaining trust between the data team and data consumers even in the very early stages. This _will_ slow you down a little, but it's faster than rebuilding trust.
* Once there are two or more people working on the data platform (e.g. Early Stage or Mid-Stage), use pull requests and code review. This helped our team leverage CI/CD during development, which cut our defect rate significantly.
* The "Implement a solid process for SQL-based modeling" is a good candidate for Early Stage. Having that been said, the primary reason that I'd move it earlier is so that practitioners have a chance to iterate before the company gets too big.
* Mid-Stage is a great time for data activation. It's usually around this size that more opportunities for leveraging data open up (e.g. enriching a CRM or marketing tool with product data). As long as you have only one or two targets in mind, the pricing can still be reasonable.
* I'd add in some "What not to do" warnings around setting up tools that you don't need yet. For instance: metadata catalogs, augmented analytics tools (Sisu/Thoughtspot), and most entitlement tools (Immuta) will not be useful until very late growth stages, if not even later.

**Reply by adrienneyager** [reply]
Oh, and I would add using dbt Cloud to the list for Early Stage. It's a little self-serving now, but at the time it did remove the need to setup any dataflow automation tools, which save a _lot_ of time.
